Code/Framework/AzCore/Platform/Windows/AzCore/IO/Streamer/StorageDrive_Windows.cpp

```diff
@@ -242,16 +248,30 @@ namespace AZ::IO
         bool hasFinalizedReads = FinalizeReads();
         bool hasWorked = false;
 
-        if (!m_pendingReadRequests.empty())
+        TaskGraph readTasks;
+
+        // Queue as many read requests as possible in order to maximize throughput.
+        while (!m_pendingReadRequests.empty())
         {
             FileRequest* request = m_pendingReadRequests.front();
-            if (ReadRequest(request))
+            if (ReadRequest(request, readTasks))
             {
                 m_pendingReadRequests.pop_front();
                 hasWorked = true;
             }
+            else
+            {
+                break;
+            }
+        }
+
+        if (hasWorked && !readTasks.IsEmpty())
+        {
+            readTasks.Detach();
+            readTasks.SubmitOnExecutor(m_taskExecutor);
         }
```

Comment:
Is the` hasWorked` check needed here?
Could `readTasks` not be empty, but `hasWorked` be false?

Reply:
You're right, it's not needed. The `hasWorked` check can now even be more narrow, so did that as well.